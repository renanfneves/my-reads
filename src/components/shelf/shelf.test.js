import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Shelf from './shelf';

it('renders an empty shelf', () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <Shelf
      title="mockedTitle"
      books={[]}
      changeShelf={() => {}}
    />,
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});


it('renders a shelf filled with a book', () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <Shelf
      title="mockedTitle"
      books={
        [
          {
            id: 'mockedId',
            title: 'mockedTitle',
            authors: [],
            imageLinks: {
              smallThumbnail: 'mockedLink',
              thumbnail: 'mockedLink',
            },
          },
        ]
      }
      changeShelf={() => {}}
    />,
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
