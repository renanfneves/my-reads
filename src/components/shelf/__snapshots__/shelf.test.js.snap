// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a shelf filled with a book 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    mockedTitle
  </h2>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <Book
        authors={Array []}
        changeShelf={[Function]}
        id="mockedId"
        imageLinks={
          Object {
            "smallThumbnail": "mockedLink",
            "thumbnail": "mockedLink",
          }
        }
        shelf="none"
        title="mockedTitle"
      />
    </ol>
  </div>
</div>
`;

exports[`renders an empty shelf 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    mockedTitle
  </h2>
  <div
    className="bookshelf-books"
  >
    <div
      className="empty-shelf"
    >
      Empty! You should add some books here.
    </div>
    <ol
      className="books-grid"
    >
      
    </ol>
  </div>
</div>
`;
